java/statestore/src/main/java/sleeper/statestore/transactionlog/DynamoDBTransactionLogSnapshotStore.java

```diff
@@ -205,25 +205,29 @@ private static TransactionLogSnapshot getSnapshotFromItem(Map<String, AttributeV
     }
 
     public static class LatestSnapshots {
-        TransactionLogSnapshot filesSnapshot;
-        TransactionLogSnapshot partitionsSnapshot;
+        Optional<TransactionLogSnapshot> filesSnapshotOpt;
+        Optional<TransactionLogSnapshot> partitionsSnapshotOpt;
```

Comment:
I don't think the Optional class is intended to be used this way. These fields can be nullable. It also looks like they should be private and final.